Framework/Core/include/Framework/InputRecord.h

```diff
@@ -285,25 +279,27 @@ class InputRecord
         throw runtime_error("Inconsistent serialization method for extracting span");
       }
       using ValueT = typename T::value_type;
-      if (header->payloadSize % sizeof(ValueT)) {
+      auto payloadSize = DataRefUtils::getPayloadSize(ref);
+      if (payloadSize % sizeof(ValueT)) {
         throw runtime_error(("Inconsistent type and payload size at " + std::string(ref.spec->binding) + "(" + DataSpecUtils::describe(*ref.spec) + ")" +
```

Comment:
I would just use fmt::format, actually...

Reply:
did not change in the runtime error message here, its just after the change, but we can correct all that later